Tasks/DeployAzureResourceGroup/AzureUtilityGTE1.0.ps1

```diff
@@ -31,19 +31,19 @@ function Create-AzureResourceGroupIfNotExist
 function Deploy-AzureResourceGroup
 {
     param([string]$csmFile,
-          [System.Collections.Hashtable]$csmParametersObject,
+          [string]$csmParametersFile,
           [string]$resourceGroupName,
           [string]$overrideParameters)
 
     Write-Host "[Azure Resource Manager]Creating resource group deployment with name $resourceGroupName"
 
-    if (!$csmParametersObject)
+    if (!$csmParametersFile)
     {
         $finalCommand = "`$azureResourceGroupDeployment = New-AzureRMResourceGroupDeployment -Name `"$resourceGroupName`" -ResourceGroupName `"$resourceGroupName`" -TemplateFile `"$csmFile`" $overrideParameters -Verbose -ErrorAction silentlycontinue -ErrorVariable deploymentError"
     }
     else
     {
-        $finalCommand = "`$azureResourceGroupDeployment = New-AzureRMResourceGroupDeployment -Name `"$resourceGroupName`" -ResourceGroupName `"$resourceGroupName`" -TemplateFile `"$csmFile`" -TemplateParameterObject `$csmParametersObject $overrideParameters -Verbose -ErrorAction silentlycontinue -ErrorVariable deploymentError"
+        $finalCommand = "`$azureResourceGroupDeployment = New-AzureRMResourceGroupDeployment -Name `"$resourceGroupName`" -ResourceGroupName `"$resourceGroupName`" -TemplateFile `"$csmFile`" -TemplateParameterFile `$csmParametersFile $overrideParameters -Verbose -ErrorAction silentlycontinue -ErrorVariable deploymentError"
```

Comment:
Pavan - 
a) Can you talk to Modi once to understand why this approach was done initially and is there any history on why passing as object was implemented?
b) Can you ensure you can pass secure variables as override parameters properly with ConvertTo-SecureString decoration that we recommend in Task UI